pkg/gator/test/test.go

```diff
@@ -30,11 +33,54 @@ func init() {
 // options for the Test func.
 type Opts struct {
 	// Driver specific options
-	IncludeTrace bool
-	GatherStats  bool
+	IncludeTrace              bool
+	GatherStats               bool
+	EnableExternalData        bool
+	ExternalDataProviderCache *frameworksexternaldata.ProviderCache
 }
 
 func Test(objs []*unstructured.Unstructured, tOpts Opts) (*GatorResponses, error) {
+	// search for external data providers.
+	// this has to happen before creating the client, because rego.AddExternalDataProviderCache arg to the driver must have
+	// a ProviderCache already set up
+	providerCache := frameworksexternaldata.NewCache()
+	for _, obj := range objs {
```

Comment:
If having a separate flag is for security purposes, we'd also need to load in the data separately, currently providers defined in any file would be ingested.

Reply:
I think we should figure out our stance vis-a-vis security in the other comment (link below). Then this comment will either be irrelevant, or very relevant.

https://github.com/open-policy-agent/gatekeeper/pull/2789#discussion_r1204869731

Reply:
@maxsmythe Could you please give me some pointers on how to implement this? 

Should [ReadSources()](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/gator/reader/filereader.go#L15) function contain something like
```
s, err = readExternalDataSourceFiles(externalDataProviderFiles)
...
```

Reply:
If we want to load the external data provider config from its dedicated directory, then we should change [this line](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/gator/reader/filereader.go#L19) to skip over external data provider config, and then either

1. Reuse `readFiles` function like: `s, err = readFiles(externalDataProviderFiles)`
2.  Create a new function like: `readExternalDataSourceFiles(externalDataProviderFiles)`.

and then append the result to `sources`. Am I on the right path?

Reply:
Yep!

Reply:
@maxsmythe @ritazh Please take another look. I'll add the tests afterward.